### PR TITLE
Add DxEngine drawing ETW tracing for debugging and diagnostics purposes

### PR DESCRIPTION
**Pull request by miniksa**
## Summary of the Pull Request
Adds an ETW provider for tracing out operations inside the DirectX Renderer.

## References
This supports #2191 and #778 and other rendering issues.

## PR Checklist
* [x] I work here

## Detailed Description of the Pull Request / Additional comments
This declares and defines the provider with the a GUID appropriate for the namespace and adds an initial invalidation rectangle method for figuring out what is being drawn mostly to understand what could be differential.

## Validation Steps Performed
Implemented provider.
Opened real-time ETW tracing tool
Ran the Terminal and watched invalidation events appear live

### CODE REVIEW

File: src/renderer/dx/DxRenderer.cpp [hunk 1]
```diff
@@ -925,6 +943,21 @@ void DxEngine::_InvalidOr(RECT rc) noexcept
     FAIL_FAST_IF_FAILED(InvalidateAll());
     RETURN_HR_IF(E_NOT_VALID_STATE, _isPainting); // invalid to start a paint while painting.
 
+    TraceLoggingWrite(g_hDxRenderProvider,
```
Comment (zadjii-msft) [inline src/renderer/dx/DxRenderer.cpp]: Should we have the same perf thing from #4594 here?

Comment (miniksa) [reply src/renderer/dx/DxRenderer.cpp]: Ah, sure, OK.

Comment (DHowett-MSFT) [reply src/renderer/dx/DxRenderer.cpp]: TraceLoggingWrite is inert if it's not enabled -- it's only computational logic _outside_ of a TraceLoggingWrite that needs to be guarded

File: src/renderer/dx/DxRenderer.cpp [hunk 2]
```diff
@@ -82,6 +88,12 @@ DxEngine::DxEngine() :
     _chainMode{ SwapChainMode::ForComposition },
     _customRenderer{ ::Microsoft::WRL::Make<CustomTextRenderer>() }
 {
+    const auto was = _tracelogCount.fetch_add(1);
+    if (0 == was)
```
Comment (DHowett-MSFT) [inline src/renderer/dx/DxRenderer.cpp]: clever

File: src/renderer/dx/DxRenderer.cpp (hunk 1 above)
Comment (miniksa) [reply src/renderer/dx/DxRenderer.cpp]: I mean it skips all the division that I'm doing, which arguably isn't much... but it's something.

Comment (miniksa) [reply src/renderer/dx/DxRenderer.cpp]: OK no, I think the macro hides it all because it was inside the macro. Rolling back.

**Comment by miniksa** [comment]
@msftbot wait 1 minute

**Pull request merged**